### ISSUE TITLE
A way to connect preprocessing with the task, not the learner?

### ISSUE THREAD

**Issue by mstaniak** [mlr-org/mlr#2562]
As far as I know, currently, it is possible to use makePreprocWrapper function to fuse learner with preprocessing method, but not possible to fuse task with preprocessing.

Suppose I wanted to use CV to try 20 learners on a dataset and the same dataset, but preprocessed by PCA with two different numbers of PCs. If there was a way to call PCA on appropriate folds based on the task during CV, I would have to use only 3 tasks and 20 learners instead of 1 task and 60 learners. It would reduce the number of required wrappers.
I guess there are more similar use cases. 

If you don't think that this would make sense, please close.

**Comment by larskotthoff** [comment]
It sounds like what you're asking for is a task from preprocessed data -- you can already do that easily. It doesn't make sense to fuse a task with preprocessing, you can just preprocess the data and create a new task. There's no need to do that again every time with a wrapper.

**Comment by mstaniak** [comment]
When doing CV, some operations (like PCA) should be estimated on the training data and the applied to test data, and the training set and test set change as we change folds.

**Comment by mstaniak** [comment]
Same with for example target encoding. With CV, it needs to be done on on each set of folds separately. This can be done with wrapped learnes, but potentially could be simpler with wrapped tasks.

**Comment by larskotthoff** [comment]
Hmm, this sounds to me like it should be part of the resample description then, not of the task.

In any case, we don't have any plans to support this. PRs are of course welcome if you want to have a look at implementing this yourself.
